appointment/upload-documents/uploadDocuments.php

```diff
@@ -73,21 +73,142 @@ class="submit dcf-btn dcf-btn-primary"
+						information, Federal law may not protect your tax return information from further use or distribution
+						in the event these systems are hacked or breached without our knowledge. If you agree to the disclosure
+						of your tax return information, your consent is valid for the amount of time that you specify. If
+						you do not specify the duration of your consent, your consent is valid for one year from the date
+						of signature. If you believe your tax return information has been disclosed or used improperly in a
+						manner unauthorized by law or without your permission, you may contact the Treasury Inspector General
+						for Tax Administration (TIGTA) by telephone at 1-800-366-4484, or by e-mail at complaints@tigta.treas.gov .
+						While the IRS is responsible for providing oversight requirements to Volunteer Income Tax Assistance (VITA) 
+						and Tax Counseling for the Elderly (TCE) programs, these sites are operated by IRS sponsored partners who
+						manage IRS site operations requirements and volunteer ethical standards. In addition, the locations of these 
+						ites may not be in or on federal Property. </p>
+					<li class="form-radio" id="consentVirtual">
+						<label class="form-required">I am agreeing to use this site's Virtual VITA/TCE Process</label>
+						<div>
+							<div class="dcf-btn-group" data-toggle="buttons">
+								<label class="dcf-btn dcf-btn-secondary" 
+									name="consentToVirtual" 
+									ng-model="consentData.virtualConsent" 
+									uib-btn-radio="1" 
+									required>Yes
+								</label>
+								<label class="dcf-btn dcf-btn-secondary" 
+									name="consentToVirtual" 
+									ng-model="consentData.virtualConsent" 
+									uib-btn-radio="0" 
+									required>No
+								</label>
+							</div>
+						</div>
+						<div ng-show="consentForm.$submitted">
+							<label class="error" ng-show="consentForm.consentToVirtual.$error.required">This field is required.</label>
+						</div>
+					</li>
+
+					<li class="dcf-form-group form-textfield">
+						<label class="dcf-label form-label form-required" for="signature">TYPE YOUR NAME HERE</label>
+						<input type="text" class="dcf-input-text form-control" name="signature" id="signature" ng-model="consentData.signature" required>
+						<div ng-show="consentForm.$submitted || consentForm.signature.$touched">
+							<label class="error" ng-show="consentForm.signature.$error.required">This field is required.</label>
+						</div>
+					</li>
+
+					<p ng-if="consentForm.$invalid && consentForm.$submitted" class="error">Your input is invalid, please correct the errors above and re-submit this form</p>
+					<input type="submit" 
+						value="Submit" 
+						class="submit dcf-btn dcf-btn-primary dcf-mt-4"
+						ng-disabled="!consentData.virtualConsent == 1 || consentData.reviewConsent == null || consentData.signature == null || consentData.signature == ''" />
```

Comment:
Do I need to check this stuff or should I just let form validation do its work? If none of these are here, it is functional and won't allow submit, but this ensures two-way binding has a say, as well.

Reply:
I would check them--might as well have that extra guarantee. Also, I think relying on form validation obscures what we actually expect to happen when you read this HTML--so I like being explicit that for this button to work, this is what we expect